icon.c

```diff
@@ -184,16 +195,24 @@ static char *resolve_icon(struct mako_notification *notif) {
 }
 
 struct mako_icon *create_icon(struct mako_notification *notif) {
-	char *path = resolve_icon(notif);
-	if (path == NULL) {
-		return NULL;
+	GdkPixbuf *image = NULL;
+	if (notif->image_data != NULL) {
+		image = load_image_data(notif->image_data);
 	}
 
-	GdkPixbuf *image = load_image(path);
-	free(path);
 	if (image == NULL) {
```

Comment:
`} else {`

Reply:
I think we should fallback to app_icon if loading image-data fails?

Reply:
Okay